perfkitbenchmarker/traces/perfspect.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2021 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+PerfSpect is a system performance profiling and processing tool based on linux perf.
+
+Usage:
+  Required flags:
+  --perfspect
+
+  Example:
+  ./pkb.py --cloud=AWS --benchmarks=your_wl --machine_type=m5.2xlarge --os_type=ubuntu2004 --perfspect
+
+  Refer to ./perfkitbenchmarker/data/perfspect/README.md for more details on flags and usage
+"""
+
+import logging
+import os
+import posixpath
+
+from absl import flags
+from six.moves.urllib.parse import urlparse
+
+from perfkitbenchmarker import events
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker import data
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_boolean('perfspect', False,
+                     'Install and run perfspect on the target system.')
+flags.DEFINE_string('perfspect_tarball', None,
+                    'Local path to perfspect tarball.')
+flags.DEFINE_string('perfspect_url', None,
+                    'URL for downloading perfspect tarball.')
+
+PERFSPECT_ARCHIVE_URL = 'https://github.com/intel/PerfSpect/releases/download/v1.1.0/perfspect_1.1.0.tgz'
+PREREQ_UBUNTU = ['linux-tools-common',
+                 'linux-tools-generic',
+                 'linux-tools-`uname -r`']
+PREREQ_CENTOS = ['perf']
+PREREQ_PKGS = ['python3-pip']
+
+
+class PerfspectCollector(object):
+  """ Manages running telemetry during a test, and fetching the results folder. """
+
+  telemetry_dir = '/opt/perf_telemetry'
+
+  def __init__(self):
+    self.pid = None
+    self.perf_dir = None
+
+  def _InstallOSReqs(self, vm):
+    """ Installs prereqs depending on the OS """
```

Comment:
We follow Google style / [PEP-257](https://www.python.org/dev/peps/pep-0257/) for docstring. Please remove the whitespace around the quotes: 

"""Installs prereqs depending on the OS."""

instead of 

""" Installs prereqs depending on the OS """

Reply:
And similar occurrences below.